gluonnlp/data/candidate_sampler.py

```diff
@@ -34,61 +31,3 @@ class CandidateSampler(object):
 
     def __call__(self):
         raise NotImplementedError
-
-
-class UnigramCandidateSampler(CandidateSampler):
-    """Unigram Candidate Sampler
-
-    Parameters
-    ----------
-    weights : mx.nd.NDArray
-        Unnormalized class probabilities.
-
-    """
-
-    def __init__(self, weights):
-        self.N = weights.size
-        total_weights = weights.sum()
-        self.prob = (weights * self.N / total_weights).asnumpy().tolist()
-        self.alias = [0] * self.N
-
-        # sort the data into the outcomes with probabilities
-        # that are high and low than 1/N.
-        low = []
-        high = []
-        for i in range(self.N):
-            if self.prob[i] < 1.0:
-                low.append(i)
-            else:
-                high.append(i)
-
-        # pair low with high
-        while len(low) > 0 and len(high) > 0:
-            l = low.pop()
-            h = high.pop()
-
-            self.alias[l] = h
-            self.prob[h] = self.prob[h] - (1.0 - self.prob[l])
-
-            if self.prob[h] < 1.0:
-                low.append(h)
-            else:
-                high.append(h)
-
-        for i in low + high:
-            self.prob[i] = 1
-            self.alias[i] = i
-
-        # convert to ndarrays
-        self.prob = mx.nd.array(self.prob)
-        self.alias = mx.nd.array(self.alias)
-
-    def __call__(self, k):
-        """Draw k samples from the distribution."""
-        idx = mx.nd.array(np.random.randint(0, self.N, size=k))
-        prob = self.prob[idx]
-        alias = self.alias[idx]
-        where = mx.nd.random.uniform(shape=k) < prob
```

Comment:
context

Reply:
Added. Also sampling is now done on the same context with mxnet instead of using numpy on cpu.